src/java/org/apache/cassandra/journal/OnDiskIndex.java

```diff
@@ -207,27 +208,27 @@ public int[] lookUp(K id)
 
         for (int i = keyIndex - 1; i >= 0 && id.equals(keyAtIndex(i)); i--)
         {
-            int length = offsets.length;
-            offsets = Arrays.copyOf(offsets, length + 1);
-            offsets[length] = offsetAtIndex(i);
+            int length = records.length;
+            records = Arrays.copyOf(records, length + 1);
+            records[length] = recordAtIndex(i);
         }
 
         for (int i = keyIndex + 1; i < entryCount && id.equals(keyAtIndex(i)); i++)
         {
-            int length = offsets.length;
-            offsets = Arrays.copyOf(offsets, length + 1);
-            offsets[length] = offsetAtIndex(i);
+            int length = records.length;
+            records = Arrays.copyOf(records, length + 1);
+            records[length] = recordAtIndex(i);
         }
 
-        Arrays.sort(offsets);
-        return offsets;
+        Arrays.sort(records);
```

Comment:
we compare the longs unsigned in `binarySearchUnsigned`, but not here, do we even need to compare unsigned? `offset` part of this long is always a normal int

Reply:
you are right; we do not need this